soroban-sdk/src/testutils.rs

```diff
@@ -48,46 +45,54 @@ pub trait Logger {
     fn print(&self);
 }
 
+/// Test utilities for [`AccountId`][crate::accounts::AccountId].
+pub trait AccountId {
+    /// Generate a random account ID.
+    //
+    // The value filled is not cryptographically secure and is only intended for
+    // use in tests.
+    fn random(env: &Env) -> crate::AccountId;
+}
+
 /// Test utilities for [`Accounts`][crate::accounts::Accounts].
 pub trait Accounts {
     /// Generate an account ID.
-    fn generate(&self) -> AccountId;
+    fn generate(&self) -> crate::AccountId;
 
     /// Generate and account ID and creates an account.
-    fn generate_and_create(&self) -> AccountId;
+    fn generate_and_create(&self) -> crate::AccountId;
 
     /// Create an account.
-    fn create(&self, id: &AccountId);
+    fn create(&self, id: &crate::AccountId);
 
     /// Set the thresholds of an account.
-    fn set_thresholds(&self, id: &AccountId, low: u8, med: u8, high: u8);
+    fn set_thresholds(&self, id: &crate::AccountId, low: u8, med: u8, high: u8);
 
     /// Set the weight of a signer of an account.
     ///
     /// Setting a weight of zero removes the signer from the account.
-    fn set_signer_weight(&self, id: &AccountId, signer: &BytesN<32>, weight: u8);
+    fn set_signer_weight(&self, id: &crate::AccountId, signer: &crate::BytesN<32>, weight: u8);
 
     /// Remove an account.
-    fn remove(&self, id: &AccountId);
+    fn remove(&self, id: &crate::AccountId);
+}
+
+/// Test utilities for [`BytesN`][crate::BytesN].
+pub trait BytesN<const N: usize> {
+    // Generate a BytesN filled with random bytes.
+    //
+    // The value filled is not cryptographically secure and is only intended for
+    // use in tests.
+    fn random(env: &Env) -> crate::BytesN<N>;
 }
 
-/// Generates a Rust array of N random bytes.
-pub fn random_bytes_array<const N: usize>() -> [u8; N] {
+/// Generates an array of N random bytes.
+///
+/// The value returned is not cryptographically secure and is only intended for
+/// use in tests.
```

Comment:
nit: I think crypto-security comment is appropriate, but the 'only intended for use in tests' seems redundant, given that all these functions belong to the test utils IIUC and can't be used inside the contracts. Up to you though.

Reply:
I just want to emphasise that someone shouldn't import testutils to get this capability into other applications for other uses. Just being overly cautious.

Reply:
Updated in ec85e79.